src/common/src/framer_manager.cpp

```diff
@@ -0,0 +1,221 @@
+// |  COPYRIGHT NovAtel Inc, 2022. All rights reserved.                          |
+// |                                                                             |
+// |  Permission is hereby granted, free of charge, to any person obtaining a    |
+// |  copy of this software and associated documentation files (the "Software"), |
+// |  to deal in the Software without restriction, including without limitation  |
+// |  the rights to use, copy, modify, merge, publish, distribute, sublicense,   |
+// |  and/or sell copies of the Software, and to permit persons to whom the      |
+// |  Software is furnished to do so, subject to the following conditions:       |
+// |                                                                             |
+// |  The above copyright notice and this permission notice shall be included    |
+// |  in all copies or substantial portions of the Software.                     |
+// |                                                                             |
+// |  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR |
+// |  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,   |
+// |  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL    |
+// |  THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER |
+// |  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING    |
+// |  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER        |
+// |  DEALINGS IN THE SOFTWARE.                                                  |
+// |                                                                             |
+// ===============================================================================
+// ! \file framer_manager.cpp
+// ===============================================================================
+
+#include "novatel_edie/common/framer_manager.hpp"
+
+using namespace novatel::edie;
+
+FramerManager::FramerManager() : pclMyLogger(Logger::RegisterLogger("FramerManager")), pclMyCircularDataBuffer(std::make_shared<CircularBuffer>())
+{
+    pclMyCircularDataBuffer->Clear();
+    pclMyLogger->debug("Framer Manager initialized");
+}
+
+void FramerManager::RegisterFramer(const FRAMER_ID framerId_, std::unique_ptr<FramerBase> framer_, std::unique_ptr<MetaDataBase> metadata_)
+{
+    framerRegistry.emplace_back(framerId_, std::move(framer_), std::move(metadata_), 0);
+}
+
+void FramerManager::ReorderFramers()
+{
+    auto it = std::min_element(framerRegistry.begin(), framerRegistry.end(), [](const FramerElement& a, const FramerElement& b) {
+        return a.framer->uiMyFrameBufferOffset < b.framer->uiMyFrameBufferOffset;
+    });
+
+    if (it != framerRegistry.end())
+    {
+        FramerElement earliestFramerElement = std::move(*it);
+        framerRegistry.erase(it);
+        framerRegistry.push_front(std::move(earliestFramerElement));
```

Comment:
I think splice works here and removes move semantics

```suggestion
        framerRegistry.splice(framerRegistry.begin(), framerRegistry, it);
```

Reply:
deque's apparently don't have the splice functionality